sdcm/cluster_k8s/__init__.py

```diff
@@ -2084,14 +2088,15 @@ def wait_for_pods_readiness(self, pods_to_wait: int, total_pods: int):
 
     def generate_namespace(self, namespace_template: str) -> str:
         # Pick up not used namespace knowing that we may have more than 1 Scylla cluster
-        namespaces = self.k8s_cluster.kubectl(
-            "get namespaces --no-headers -o=custom-columns=:.metadata.name").stdout.split()
-        for i in range(1, len(namespaces)):
-            candidate_namespace = f"{namespace_template}{'-' + str(i) if i > 1 else ''}"
-            if candidate_namespace not in namespaces:
-                self.k8s_cluster.kubectl(f"create namespace {candidate_namespace}")
-                return candidate_namespace
-        raise RuntimeError("No available namespace for a Scylla cluster was found")
+        with Lock():
```

Comment:
what's this lock is doing? seem like you creating new one each time, 
i.e. I'm not sure it really prevent it being called multiple time from multiple threads or processes...

Reply:
We run creation of clusters in parallel. So, we must use lock mechanism here to avoid race.

Reply:
O.k. you need a lock, but you are creating the lock again and again, hence in each entry to the function it's not the same lock being used. you need to create it on some shared location, globally or on some object.

Reply:
That's really good catch. I wonder how I made such a mistake. Thanks.
Will fix it.

Reply:
Done